src/TimeFrameSettings/EditTimeFrameForm.tsx

```diff
@@ -23,39 +26,92 @@ export class EditTimeFramesForm extends React.Component<IEditTimeFramesProps, IE
+                            stateContext.actions.toggleSettings({ expanded: false });
+                        },
+                        disabled: addBoxOpen 
+                    }
+                ]}
+                onDismiss={() => {
+                    stateContext.actions.toggleSettings({ expanded: false });
+                }}>
+                <div className={"time-frame-settings"}>
+                    <div className={"time-frame-settings-header"}>{"Add new time frames below and select which will be the current time frame."}</div>
+                    <div>
+                        <RadioButtonGroup onSelect={this.changeCurrentTimeFrame} selectedButtonId={draftTimeFrameSet.currentTimeFrameId}>
+                            {this.state.draftTimeFrameSet.timeFrames.map((value, index) => (
+                                <RadioButton className={"time-frame-box"} id={value.id}>
+                                    <EditTimeFrame item={value} saveTimeFrame={this.updateName} />
+                                </RadioButton>
+                            ))}
+                        </RadioButtonGroup>
+                        {addBoxOpen && addTimeFrameBox}
+                        <Button subtle={true} text={"Add new time frame"} iconProps={{ iconName: "Add" }} onClick={() => this.setState({ addBoxOpen: true })} disabled={addBoxOpen} />
+                    </div>
+                </div>
+            </Panel>
 
-        return (
-            <div>
-                {items.map((value, index) => (
-                    <EditTimeFrame item={value} saveTimeFrame={updateTimeFrame} canBeEdited={!this.state.isSomeoneBeingEdited} editingCallback={updateIfAnyCellIsInEditMode} isEditing={false} />
-                ))}
-                {this.state.addBoxOpen && addTimeFrame}
-                <Button subtle={true} text={"Add new time frame"} iconProps={{ iconName: "Add" }} onClick={() => this.setState({ addBoxOpen: true })} disabled={this.state.addBoxOpen} />
-            </div>
         )
     }
+
+    private closeAddBox = () => {
+        this.setState({ addBoxOpen: false });
+    }
+
+    private addTimeFrame = (tf) => {
+        const newSet = produce(this.state.draftTimeFrameSet, draft => {
+            draft.timeFrames.push(tf);
+        });
+        this.setState({ draftTimeFrameSet: newSet, addBoxOpen: false });
+    }
+
+    private updateName = (item: TimeFrame, newName: string) => {
+        const newTimeFrames = produce(this.state.draftTimeFrameSet.timeFrames, draft => {
+            let editedItem = draft.find((tf) => { return tf.id === item.id });
+            editedItem.name = newName; 
+        });
+
+        const newState = {
+            currentTimeFrameId: this.state.draftTimeFrameSet.currentTimeFrameId,
+            id: this.state.draftTimeFrameSet.id,
+            timeFrames: newTimeFrames
+        }
+        this.setState({ draftTimeFrameSet: newState });
```

Comment:
@Nianwens This component doesn't recognize a new state here, even though I've made a new object. I'm not seeing why this isn't working.... I think I'm missing something obvious and I need to look at it with fresh eyes. Do you see it?

Reply:
Nevermind, I figured it out. No problem with the state update - the radio boxes are overriding my click function and setting state without the name update :)